edx_rest_framework_extensions/middleware.py

```diff
@@ -31,19 +31,29 @@ def _add_missing_jwt_permission_classes(self, view_class):
         Adds permissions classes that should exist for Jwt based authentication,
         if needed.
         """
-        view_permission_classes = getattr(view_class, 'permission_classes', tuple())
+        view_permissions = list(getattr(view_class, 'permission_classes', []))
 
-        for perm_class in self._required_permission_classes:
+        # Not all permissions are classes, some will be ConditionalPermission
+        # objects from the rest_condition library. So we have to crawl all those
+        # and expand them to see if our target classes are inside the
+        # conditionals somewhere.
+        permission_classes = []
+        while view_permissions:
+            permission = view_permissions.pop()
+            if not hasattr(permission, 'perms_or_conds'):
+                permission_classes.append(permission)
+            else:
+                for child in getattr(permission, 'perms_or_conds', []):
```

Comment:
Is there any public API or interface on `ConditionalPermission` that we can use?  Do you think this assumes an implementation of it that may change over time?  Or is this part of its contract?

Reply:
It's a public attribute. I don't know if it's really promised in their API, as their docs are pretty minimal (just a usage in their repo's README. But I'm adding the test now, so if they break compatibility in a later version, builds for this repo should start failing.